frontend/src/components/dashboard/hooks/useDissimilarAnalyses/index.ts

```diff
@@ -69,33 +45,98 @@ const getDissimilarAnalyses = async <T extends CurrentAnalysis>(
 
   await Promise.all(promises);
 
-  let maxDistance = 0;
-  let bestCombination: number[] = new Array(numberOfAnalyses)
+  return distances;
+};
+
+const getBestCombination = (
+  numberOfAnalyses: number,
+  distances: number[][],
+  initialDistance: number = 0,
+  combineDistances: (overallDistance: number, distance: number) => number = (
+    d1,
+    d2,
+  ) => d1 + d2,
+  isNewDistanceBetter: (bestDistance: number, newDistance: number) => boolean,
+): number[] => {
+  let bestDistance = 0;
+  // store the "best" combination in the form of indices of the analyses
+  let bestCombination: number[] = new Array<number>(numberOfAnalyses)
     .fill(0)
     .map((_, idx) => idx);
 
   // compute all combinations of k elements from the analysis set
   for (const combination of getCombinationsUnsafe(
-    analyses.length,
+    distances.length,
     numberOfAnalyses,
   )) {
-    let overallDistance = 0;
+    let newDistance = initialDistance;
     // then, for each pair of analyses, add the distance
     for (const pair of getCombinationsUnsafe(combination.length, 2)) {
       const [i, j] = pair.map((index) => combination[index]);
-
-      overallDistance += distances[i][j];
+      newDistance = combineDistances(newDistance, distances[i][j]);
     }
 
-    if (overallDistance > maxDistance) {
-      maxDistance = overallDistance;
+    if (isNewDistanceBetter(bestDistance, newDistance)) {
+      bestDistance = newDistance;
       // store a copy of the combination with the maximum distance,
       // because getCombinationsUnsafe will otherwise override combination.
       bestCombination = [...combination];
     }
   }
 
-  return bestCombination.map((idx) => analyses[idx].analysis);
+  return bestCombination;
+};
+
+export const maximizeMinimumDistance = (
+  numberOfAnalyses: number,
+  distances: number[][],
+): number[] =>
+  getBestCombination(
+    numberOfAnalyses,
+    distances,
+    // since we take the minimum distance, we need to start with the maximum possible value
+    Number.POSITIVE_INFINITY,
+    (d1, d2) => Math.min(d1, d2),
+    // the new distance is better if it is greater than the best distance (maximizing)
+    (bestDistance, newDistance) => bestDistance < newDistance,
+  );
+
+const getDissimilarAnalyses = async <T extends CurrentAnalysis>(
+  analysesIn: T[] | undefined,
+  numberOfAnalyses: number,
+  distanceType: DistanceType = DistanceType.pqGrams,
```

Comment:
I would consider the `DistanceMaximizationType` as an extra parameter, for now supporting only `DistanceMaximizationType.MinimumDistance`, though I'd argue we could trivially keep the old logic around for future reference.